main.nf

```diff
@@ -232,11 +239,46 @@ process star {
           --outWigType bedGraph \\
           --outSAMtype BAM SortedByCoordinate \\
           --readFilesCommand zcat \\
-          --outFileNamePrefix $prefix
+          --outFileNamePrefix \$prefix
      """
 }
 
 
+//Function that checks the alignment rate of the STAR output
+//and returns true if the alignment passed and otherwise false
+
+def check_log(logs) {
+    def percent_aligned = 0;
+    logs.eachLine { line ->
+    if ((matcher = line =~ /Uniquely mapped reads %\s*\|\s*([\d\.]+)%/)) {
+                percent_aligned = matcher[0][1]
+            }
+    }
+    percent_aligned.toFloat()
+    if(percent_aligned.toFloat() <='10'.toFloat() ){
+        println "#################### VERY POOR ALIGNMENT RATE ONLY ${percent_aligned}%! FOR ${logs}"
+        false
+    } else {
+	println "Passed aligment with ${percent_aligned}%! FOR ${logs}"
+        true
+   }
+}
+
+//Filter removes all 'aligned' chanels that fail the check
+ 
+aligned.filter { logs, bams -> check_log(logs) }
+    .map {  logs, bams -> bams }
+    .set {SPLIT_BAMS }
+    SPLIT_BAMS.into {bam_count; bam_rseqc; bam_preseq; bam_markduplicates; bam_featurecounts; bam_stringtieFPKM}
+
+//Counts the number of bam files from STAR. Uses this to determine whether to run 'sampleCorrelation' process or not
+bam_count.count()
+    .subscribe { if (count.toFloat() <= '3'.toFloat()){
+                    params.sampleLevel = true
```

Comment:
This is nice, but if this has been specified on the command line it will be overwritten. Need to wrap in an extra if statement that checks whether it's been set.